compysition/actors/util/udplib.py

```diff
@@ -178,15 +178,15 @@ def run(self):
     
     def reap_peers(self):
         now = time.time()
-        for peer in list(self.peers.values()):
-            if peer.expires_at < now:
-                self.peers.pop(peer.uuid)
-                self.log("Reaping expired peer ({0}) from '{1}' peer pool. (Total remaining: {2})".format(peer.uuid, self.service, len(self.peers) + 1))
+        remove = [peer for peer in self.peers.itervalues() if peer.expires_at < now]
```

Comment:
The issue here is the pop.  There may be a way to use a generator, but an iterator could fail.